### PR TITLE
[SDSP-2065] CI pipeline improvements

### PR DESCRIPTION
**Pull request by st0ck**
Added
- ruby files linter
- ruby tests job

Rubocop config defaults are still to be decided.

Jira ticket: https://adgear.atlassian.net/browse/SDSP-2065

[SDSP-2065]: https://adgear.atlassian.net/browse/SDSP-2065?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/lock_test.rb [hunk 1]
```diff
@@ -30,9 +30,9 @@ def test_lint
   end
 
   def test_version
-    major, minor, patch = Resque::Version.split('.')
-    assert_equal 1, major.to_i
-    assert minor.to_i >= 17
+    version = Object.const_defined?('Resque::Version') ? Resque::Version : Resque::VERSION
+    major, minor, _patch = version.split('.')
+    assert(minor.to_i >= 17) if major.to_i == 1
```
Comment (newbermtl) [inline test/lock_test.rb:35]: If the major version is no longer one then the assertion will not run, wouldn't it be better to have a separate assertion for the major version?

Comment (st0ck) [reply test/lock_test.rb:35]: As I see it, the test checks that the bundler version is at least 1.17. Potential maintainers will get into trouble if they use the bundler version < 1. Do you think we need a test for this case?

Comment (newbermtl) [reply test/lock_test.rb:36]: Ahh ok I understand the logic. I was wondering why we don't test for an increase in version but if all we want to ensure is that the version is at least 1.17 then this makes sense

**Review by newbermtl — APPROVED** [review]
LGTM 👍

**Pull request merged**